app/lib/frontend/search_service.dart

```diff
@@ -62,11 +70,65 @@ class SearchService {
   /// The Custom Search client API stub.
   final customsearch.CustomsearchApi csearch;
 
-  SearchService(this.httpClient, this.csearch);
+  /// The HTTP client used for making calls to our search service.
+  final http.Client searchServiceClient;
+
+  SearchService(this.httpClient, this.csearch, this.searchServiceClient);
 
   /// Search for packes using [queryText], starting at offset [offset] returning
   /// max [numResults].
-  Future<SearchResultPage> search(SearchQuery query) async {
+  Future<SearchResultPage> search(SearchQuery query, bool useService) async {
+    if (useService) {
+      try {
+        final SearchResultPage page = await _searchService(query);
+        if (page != null) return page;
```

Comment:
Maybe add a logging call after the if, so we know when we fall back to CSE in the case where no exception is raised.